src/lib/components/compositions/dialog/dialog.svelte

```diff
@@ -41,23 +44,54 @@
 </script>
 
 <!-- Inspired by https://github.com/mvolkmann/svelte-dialog -->
-<!-- TODO: Animate modal from bottom and fade in -->
 <dialog
   id={dialogId}
   bind:this={dialogEl}
-  class={`tw_dialog-box dark:tw_dialog-box_dark md:tw_dialog-box_medium`}
+  class={`tw_dialog-box dark:tw_dialog-box_dark md:tw_dialog-box_medium ${
+    forceOpen ? '' : 'tw_dialog-box_animate'
+  }`}
+  open={forceOpen}
 >
-  <div class="tw_w-full tw_-mt-4 tw_relative tw_text-right">
-    <button
-      class="tw_dialog-x tw_focusable-element dark:tw_focusable-element_dark"
-      on:click={() => closeDialog({ reason: 'user' })}
-      aria-controls={dialogId}
-    >
-      <XIcon
-        classes="tw_inline-block tw_fill-current tw_text-secondary-dark dark:tw_text-secondary-light"
-        ><T key="closeModal" /></XIcon
+  {#if withHeader}
+    <div class="tw_dialog-header dark:tw_dialog-header_dark">
+      <div
+        class="tw_dialog-logo dark:tw_dialog-logo_dark"
+        style={`--logo-dark: url("${$style?.logo?.dark}"); --logo-light: url("${$style?.logo?.light}"); ${
+          $style?.logo?.height ? `height: ${$style?.logo.height}px;` : ''
+        } ${$style?.logo?.width ? `width: ${$style?.logo.width}px;` : ''}`}
+      />
+      <button
+        class="tw_dialog-x md:tw_dialog-x_medium tw_focusable-element dark:tw_focusable-element_dark"
+        on:click={() => closeDialog({ reason: 'user' })}
+        aria-controls={dialogId}
       >
-    </button>
+        <XIcon
+          classes="tw_inline-block tw_fill-current tw_text-secondary-dark dark:tw_text-secondary-light"
+          ><T key="closeModal" /></XIcon
+        >
+      </button>
```

Comment:
After looking at this, we could probably separate this out into it's own x-button component. Not sure I want to take this on at this moment though.

Reply:
Just for my own knowledge, what do you mean by `x-button` and why / when would we consider refactoring this to an `x-button`?